### DISCUSSION TITLE
Why I can connect to my public server when its firewall is closed

### DISCUSSION THREAD

**Discussion by lewis617** [jamulussoftware/jamulus#2171] [Technical Support & Help] [answered]
I ran a jamulus server on cloud, when I made it public, I could connect to it without opening the firewall(security group).  why? how did it work?

plus, when I made my server private, I had to open firewall, otherwise, I couldn't connect to it.

**Comment by passing** [comment, marked as answer]
Directory servers do "hole punching" (<https://en.wikipedia.org/wiki/Hole_punching_(networking)>) for the servers connected to them.
